Fix for the dependency matching in Makefile.deps

zenoamaro opened the pull request:
This lets ino match every dependency (and not only the first) in `Makefile.deps` in lines containing multiple dependencies. This should also fix issue #13.

```
sketch.o: .build/src/sketch.cpp \
    /Applications/Arduino.app/.../arduino/Arduino.h \
    /Applications/Arduino.app/.../arduino/binary.h \
    ...
    /Applications/Arduino.app/.../arduino/new.h \
    /Applications/Arduino.app/.../standard/pins_arduino.h \
    lib/SSegs/SSegs.h lib/Framebuffer/Framebuffer.h lib/Metro/Metro.h
```

Fixed swapping _match_ for _findall_ during `_scan_dependencies` (unfortunately named group references cannot be used anymore as _findall_ returns tuples).

I didn't see test covering this, so I didn't update them.

zenoamaro committed:
fix #13: fixes matching only the first dependency per line in Makefile.deps

fabiankr commented:
Thanks for the fix. Unfortunately, I'm still getting compiler errors when adding a simple "Defines.h" in an empty sketch. Where do you put the header file in your configuration? Could you maybe upload an example sketch so we can solve this issue?

zenoamaro committed:
fixes #13: added "src" to the inclusion paths; changes in the dependency scan

- Added `src/` to the build inclusion paths, but not recursively.
- Now we split multiple includes in a line when parsing
`dependencies.d` prior to matching.
- Libs can still be found under whatever subdirectory of `lib/`

zenoamaro committed:
Don't call gcc when there are no sources

zenoamaro committed:
Now we abort on make errors; hidden superfluous *Nothing to be done* messages

zenoamaro committed:
Added a test sketch

fabiankr commented:
It works now, thank you.
